packages/flutter/lib/src/widgets/animated_size.dart

```diff
@@ -49,21 +50,23 @@ import 'framework.dart';
 /// See also:
 ///
 ///  * [SizeTransition], which changes its size based on an [Animation].
-class AnimatedSize extends SingleChildRenderObjectWidget {
-  /// Creates a widget that animates its size to match that of its child.
-  ///
-  /// The [curve] and [duration] arguments must not be null.
+class AnimatedSize extends StatefulWidget {
   const AnimatedSize({
     Key? key,
-    Widget? child,
+    this.child,
     this.alignment = Alignment.center,
     this.curve = Curves.linear,
     required this.duration,
     this.reverseDuration,
-    required this.vsync,
+    @Deprecated('Ignored') TickerProvider? vsync,
```

Comment:
nit: these deprecation notices need to follow a certain pattern, see "Breaking Changes" in the wiki.

Reply:
done